synapse/http/federation/srv_resolver.py

```diff
@@ -152,6 +153,10 @@ async def resolve_service(self, service_name: bytes) -> List[Server]:
                 return list(cache_entry)
             else:
                 raise e
+        except defer.TimeoutError as e:
+            raise defer.TimeoutError(
```

Comment:
I'm not sure if this exception should raise "all the way" here via this, but at least it has different wording.